pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DateTimeFunctions.java

```diff
@@ -715,4 +730,41 @@ public static String dateTimeConvert(String timeValueStr, String inputFormatStr,
       return new DateTimeFormatSpec(outputFormatStr).fromMillisToFormat(roundedTimeValueMs);
     }
   }
+
+  @ScalarFunction(names = {"timestampAdd", "dateAdd"})
+  public static long timestampAdd(String unit, int interval, long timestamp) {
+    ISOChronology chronology = ISOChronology.getInstanceUTC();
+    long millis = getTimestampField(chronology, unit).add(timestamp, interval);
+    return millis;
+  }
+
+  @ScalarFunction(names = {"timestampDiff", "dateDiff"})
+  public static long timestampDiff(String unit, long timestamp1, long timestamp2) {
+    ISOChronology chronology = ISOChronology.getInstanceUTC();
+    long millis = getTimestampField(chronology, unit).getDifferenceAsLong(timestamp1, timestamp2);
```

Comment:
The return is not `millis` but value in `unit`. Also, seems the argument is in wrong order
```suggestion
    return getTimestampField(chronology, unit).getDifferenceAsLong(timestamp2, timestamp1);
```

Reply:
done